core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.component.ts

```diff
@@ -268,6 +273,33 @@ export class TopNavigationBarComponent implements OnInit, OnDestroy {
     }, 0);
   }
 
+  ngAfterViewChecked(): void {
+    if (this.CLASSROOM_PROMOS_ARE_ENABLED) {
+      this.accessValidationBackendApiService.validateAccessToClassroomPage(
+        'math').then(()=>{
+        this.classroomBackendApiService.fetchClassroomDataAsync(
+          'math').then((classroomData) => {
+          this.classroomData = classroomData.getTopicSummaries();
+        });
+      });
+    }
+
+    this.learnDropdownOffset = this.getDropdownOffset(
+      '.learn-tab', (this.CLASSROOM_PROMOS_ARE_ENABLED) ? 688 : 300);
+    // https://stackoverflow.com/questions/34364880/expression-has-changed-after-it-was-checked
+    this.changeDetectorRef.detectChanges();
+  }
+
+  getDropdownOffset(cssClass: string, width: number): number {
```

Comment:
Can you please add a descriptive comment about why this is required?

Reply:
Done!